primitives/runtime/src/offchain/storage.rs

```diff
@@ -28,6 +28,35 @@ pub struct StorageValueRef<'a> {
 	kind: StorageKind,
 }
 
+/// Result of asking for a particular key from the store.
+#[derive(Debug, PartialEq, Eq)]
+pub enum StorageEntry<T> {
+	/// Successfully retrieved the stored value.
+	Occupied(T),
+	/// Successfully looked up the key but noting is currently stored at that location.
+	Vacant,
+	/// There is no entry in the db for that key.
+	KeyNotStored,
+}
+
+/// Possible errors when mutating a storage value.
+#[derive(Debug, PartialEq, Eq)]
+pub enum MutateStorageError<T, E> {
+	/// The new value might not encode correctly or the underlying db failed to update.
+	CalculatedButNotStored(T),
+	/// The function given to us to create the value to be stored failed.
+	FailedToCalculateNewValue(E)
```

Comment:
This enum still feels like the wording could be tighter. Bike shed away!

Reply:
```suggestion
	ValueFunctionFailed(E)
```